### PR TITLE
Add `JsonMissingBlock` Check

### PR DESCRIPTION
**Pull request by navdeep5**
## What are you adding in this PR?

Addresses https://github.com/Shopify/shopify/issues/552694

This PR introduces enhanced validation checks for JSON template files, specifically ensuring that:
1) Block types exist as valid files in the `blocks/` dir
2) Block types are allowed in their respective `section` or `block` at the root-level

image
image
image

## What's next? Any followup issues?

With this being a new check, we will want to update https://github.com/Shopify/shopify-dev so users can understand their errors. Developer Docs changes: https://github.com/Shopify/shopify-dev/pull/51342

## What did you learn?

I have never worked or come across JSON template files before so this was fun to learn about.

## Before you deploy

- [x] This PR includes a new checks or changes the configuration of a check
  - [x] I included a minor bump `changeset`
  - [x] It's in the `allChecks` array in `src/checks/index.ts`
  - [x] I ran `yarn build` and committed the updated configuration files
    
    
    
    - [ ] If applicable, I've updated the `theme-app-extension.yml` config

- [x] I included a minor bump `changeset`
- [x] My feature is backward compatible

- [ ] I included a patch bump `changeset`

### CODE REVIEW

**Review by miazbikowski — APPROVED** [review]
We have that issue with `onCodePathEnd` checks not running when the editor opens, but rather on refocusing on the file tab, but that's not strictly related to this check. Otherwise, I tophatted and code lgtm.

File: packages/theme-check-common/src/checks/json-missing-block/index.spec.ts [hunk 1]
```diff
@@ -0,0 +1,378 @@
+import { expect, describe, it } from 'vitest';
+import { JSONMissingBlock } from './index';
+import { check, MockTheme } from '../../test';
+
+describe('Module: JsonMissingBlock', () => {
+  describe('File existence validation', () => {
+    it('should report an offense when a block does not exist', async () => {
+      const theme: MockTheme = {
+        'templates/failing.json': `{
```
[the remark this answers is not in the document]
Comment (navdeep5) [reply packages/theme-check-common/src/checks/json-missing-block/index.spec.ts]: Oh, thanks for the heads up!

**Review by albchu — APPROVED** [review]
Change LGTM and tophats as expected! Great work Nav! I added one minor nit but feel free to _ingore_ 😏

File: packages/theme-check-common/src/checks/json-missing-block/index.spec.ts [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+            }
+          }
+        }`,
+      };
+
+      const offenses = await check(theme, [JSONMissingBlock]);
+      expect(offenses).to.be.empty;
+    });
+
+    it('should ingore templates with local blocks', async () => {
```
[the remark this answers is not in the document]
Comment (navdeep5) [reply packages/theme-check-common/src/checks/json-missing-block/index.spec.ts]: Good catch!

**Pull request merged**